usefulUtilities/spawnPointZone/entityResources/entityClientScripts/spawnPointZone.js

```diff
@@ -34,6 +36,43 @@
                 print("ERROR: COULD NOT GET ZONE USER DATA");
                 return;
             }
+            if (!spawnAreaProperties.position || Object.keys(spawnAreaProperties.position).length !== NUMBER_OF_AXES 
+                || !spawnAreaProperties.dimensions || Object.keys(spawnAreaProperties.dimensions).length !== NUMBER_OF_AXES) {
+                if (DEBUG) {
+                    print("INCORRECT POSITION OR DIMENSIONS SPECIFIED");
+                }
+                return;
+            }
+            var shouldMove = true;
+            if (spawnAreaProperties.usernameWhitelist && Array.isArray(spawnAreaProperties.usernameWhitelist)) {
+                (spawnAreaProperties.usernameWhitelist).forEach(function (newUsername) {
+                    if (newUsername.toLowerCase() === AccountServices.username.toLowerCase()) {
+                        if (DEBUG) {
+                            print("USER NAME ", newUsername, " IS ON WHITELIST. RETURNING", newUsername);
+                        }
+                        shouldMove = false;
+                    }
+                });
+            }
+            if (shouldMove) {
+                _this.moveUser(spawnAreaProperties);
+            }
+        },
+
+        moveUser: function(spawnAreaProperties) {
+            if (spawnAreaProperties.avatarRotation && Object.keys(spawnAreaProperties.avatarRotation).length === NUMBER_OF_AXES) {
+                if (DEBUG) {
+                    print("SPECIFIED AVATAR ROTATION: ", JSON.stringify(spawnAreaProperties.avatarRotation));
+                }
+                MyAvatar.orientation = Quat.fromVec3Degrees(spawnAreaProperties.avatarRotation);
+            } else {
+                var randomYDegrees = Math.random() * DEGREES_ON_AXIS;
```

Comment:
Maybe do Math.floor() on this.  Not sure if getting something like 90.5 matters.  No need to change though.

Reply:
oh, it won't matter in this case. Good catch tho